src/fluree/db/conn/file.cljc

```diff
@@ -68,14 +44,13 @@
                         (when branch (str "/" branch))
                         (str "/" type-dir "/")
                         hash ".json")
-        write-path (str (fs/local-path storage-path) "/" path)]
-    (log/debug (str "Writing " (name data-type) " at " write-path))
-    (fs/write-file write-path bytes)
+        ;; TODO: build migration to remove .json suffix so we can use :content-address? true
```

Comment:
We need non-content addressed writes for the ledger head currently. 

In order to handle this going forward, we'd need dual implementations of most of the store methods:
- FileStore
- ContentAddressedFileStore
- MemoryStore
- ContentAddressedMemoryStore
- etc.

The content-addressed versions of the store can be implemented in terms of the non-content addressed stores. The one exception is IPFS, which would have the IPFSStore and the IPNSSTore, where the IPFSStore would be used to write the commits and the IPNSStore would be used to write the head.

I'm not as opposed to having a boolean option meant for control flow, as it will save writing a ton of boilerplate, but I don't feel very strongly about it.

Reply:
> We need non-content addressed writes for the ledger head currently.
> 
> In order to handle this going forward, we'd need dual implementations of most of the store methods:

Potentially, but I think this part is the concern of the NameService. It is possible the NameService does need its own way of persisting - but it is also possible some nameservices are at a location different than data storage (e.g. Etherium as a  NameService, but IPFS as storage)

It is true that someone using FileStore will likely want the FileStore Nameservice and then both do write... and likewise IPFS will want to use IPNS for Nameservice (which relies on IPFS files) so two purposes.

But I'd rather see the NameServices implement their own methods for how they store... and that be abstracted from the protocol which looks more like `-lookup`, `-update`, etc.